### PR TITLE
Set last-modified time on exported files in Windows.

### PR DESCRIPTION
**Pull request by dbrant**
Technically the Windows runtime libraries provide a `utime` function. However, the native `SetFileTime` function actually allows us to set the modified as well as the creation time of the file, which is even better.

### CODE REVIEW

File: BasiliskII/src/Windows/posix_emu.cpp [hunk 1]
```diff
@@ -1123,3 +1123,28 @@ int my_write( int fd, const void *buffer, unsigned int count )
 	D(bug("write(%ld,%08x,%ld) = %d\n", fd, buffer, count, result));
 	return result;
 }
+
+FILETIME getFileTime(time_t time) {
```
Comment (asvitkine) [inline BasiliskII/src/Windows/posix_emu.cpp]: Please use a consistent naming convention with the rest of the file - e.g. get_file_time().

Also, make this function static if you're not using it outside this file.

File: BasiliskII/src/Windows/posix_emu.cpp [hunk 2]
```diff
@@ -1123,3 +1123,28 @@ int my_write( int fd, const void *buffer, unsigned int count )
...
+	result += time;
+	result *= 10000000LL;
+	ft.dwHighDateTime = (result >> 32);
+	ft.dwLowDateTime = (result & 0xFFFFFFFF);
+	return ft;
+}
+
+int my_utime( const char *path, struct my_utimbuf * my_times )
+{
+	auto tpath = tstr(path);
```
Comment (asvitkine) [inline BasiliskII/src/Windows/posix_emu.cpp]: Don't use auto since the project doesn't currently require C++11 to build.

**Comment by asvitkine** [comment]
Thanks, looks good - just a few comments.

File: BasiliskII/src/Windows/posix_emu.cpp (hunk 1 above)
Comment (dbrant) [inline BasiliskII/src/Windows/posix_emu.cpp]: Done

File: BasiliskII/src/Windows/posix_emu.cpp (hunk 2 above)
Comment (dbrant) [inline BasiliskII/src/Windows/posix_emu.cpp]: It looks like you use `auto` in several other places in this file. I simply copied the usage pattern from one of the other places. I'll be happy to remove it, but wanted to confirm.

Comment (dbrant) [inline BasiliskII/src/Windows/posix_emu.cpp]: It looks like you use `auto` in several other places in this file. I simply copied the usage pattern from one of the other places. I'll be happy to remove it, but wanted to confirm.

Comment (asvitkine) [inline BasiliskII/src/Windows/posix_emu.cpp]: Ah, fair enough. In that case, it's fine.

**Pull request merged**